web/src/popover_menus_data.ts

```diff
@@ -217,6 +218,22 @@ export function get_actions_popover_content_context(message_id: number): ActionP
     const should_display_read_receipts_option = realm.realm_enable_read_receipts && not_spectator;
     const should_display_remind_me_option = not_spectator;
 
+    const should_display_message_report_option = (): boolean => {
+        if (page_params.is_spectator) {
+            return false;
+        }
+        if (realm.realm_moderation_request_channel_id === -1) {
+            return false;
+        }
+        if (!current_user.is_guest) {
+            return true;
+        }
```

Comment:
Can you point me at where we decided that guests should not be able to report messages? I'd like us to document the reasoning in a comment here. 

We will also want that information to live in the /help/ documentation for reporting messages.

Oh, I see -- you're trying to implement the ACL that they can, if they're subscribed? I think actually you can delete these checks, then -- the web app won't have access to any messages the guest can't access, so these checks should never fail -- it's the server's job to enforce these rules.

At least if the rule is "You can report any message you can access", which seems fair, we should just put that in a comment here and `return true;`.

Reply:
I'd add a period at the end of the sentence, but otherwise this is great :)